pkg/validation/spec/info.go

```diff
@@ -89,15 +89,20 @@ func (e Extensions) GetObject(key string, out interface{}) error {
 	return nil
 }
 
-func (e Extensions) sanitize() {
+// Sanitize sanitizes the Extensions map such that only keys with the
+// "x-" prefix are preserved
+func (e Extensions) Sanitize() {
```

Comment:
I'm not sure how hard we should work to avoid exporting these beyond this module, but a possible alternative to exporting these might be to put them in the internal package as functions and invoke them from here and the v3 package.

Reply:
Yeah that was the alternative I was thinking of as well. We don't necessarily need the functions as a member of the corresponding types since they're not really accessing any private fields.

Reply:
If `sanitize` is the only motivator behind this reshuffling, I think -implementing it in both packages isn't terrible. It deletes keys prefixed with `x-`....

Reply:
One drawback I see to that is it may break backwards compatibility since I  imagine `spec.Schema` and `spec3.OpenAPI` are imported in quite a few places in client libraries/etc.